### PR TITLE
:bug: If the Diagram is Already Opened, Navigate to it.

### LINKED ISSUE #959: 🐛 When Opening a Diagram via File Association That Is Already Opened, the BPMN-Studio Will Not Display It Automatically
When opening a diagram via file association that is already opened, the bpmn-studio will not display it automatically.

### PR DESCRIPTION
**Pull request by Paulomart**
**Changes:**

1. Navigate to diagram if its already opened.

**Issues:**

Closes #959

PR: #968 

---

See the [examples](#example) for inspiration.

## How can others test the changes?

> Describe how others can test your changes (you can remove this section for typo fixes etc.)

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️     |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️     |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: src/modules/solution-explorer-services/SingleDiagramsSolutionExplorerService.ts [hunk 1]
```diff
@@ -39,6 +39,20 @@ export class SingleDiagramsSolutionExplorerService implements ISolutionExplorerS
     return this._openedDiagrams;
   }
 
+  /**
+   * Gets the single diagram with the given uri, if the diagram was opened
+   * before.
+   */
+  public getOpenedDiagramByURI(uri: string): IDiagram | null {
+    const indexOfDiagram: number = this._findOfDiagramWithURI(uri);
+    if (indexOfDiagram < 0) {
```
Comment (SteffenKn) [inline src/modules/solution-explorer-services/SingleDiagramsSolutionExplorerService.ts]: Extract Business Rule

File: src/modules/solution-explorer-services/SingleDiagramsSolutionExplorerService.ts [hunk 2]
```diff
@@ -39,6 +39,20 @@ export class SingleDiagramsSolutionExplorerService implements ISolutionExplorerS
...
+      return null;
+    }
+
+    const diagramWithURI: IDiagram = this._openedDiagrams[indexOfDiagram];
+    return diagramWithURI;
```
Comment (SteffenKn) [inline src/modules/solution-explorer-services/SingleDiagramsSolutionExplorerService.ts]: Add some spacing

File: src/modules/solution-explorer/solution-explorer-panel/solution-explorer-panel.ts [hunk 3]
```diff
@@ -145,6 +145,12 @@ export class SolutionExplorerPanel {
       await this._navigateToDetailView(openedDiagram);
 
     } catch (error) {
+      // The diagram may already be opened.
+      const diagram: IDiagram | null = await this.solutionExplorerList.getOpenedSingleDiagramByURI(uri);
+      if (diagram !== null) {
```
Comment (SteffenKn) [inline src/modules/solution-explorer/solution-explorer-panel/solution-explorer-panel.ts]: Extract Business Rule

File: src/modules/solution-explorer-services/SingleDiagramsSolutionExplorerService.ts (hunk 2 above)
Comment (cmg-dev) [reply src/modules/solution-explorer-services/SingleDiagramsSolutionExplorerService.ts]: The reason for this is to create more contrast to spot the important things faster.

**Pull request merged**